viewer/FractalShading.js

```diff
@@ -64,18 +106,144 @@ const COLOR_BY_DISTANCE = new Cesium.CustomShader({
     fragmentShaderText: `
     void fragmentMain(FragmentInput fsInput, inout czm_modelMaterial material) {
         float dist_from_center = length(fsInput.attributes.positionMC);
-        float freq = 8.0;
+        float freq = 4.0;
         float wave = 0.5 + 0.5 * cos(2.0 * czm_pi * freq * dist_from_center);
         material.diffuse *= wave;
     }
     `
 });
 
+const COLOR_OCTANTS = new Cesium.CustomShader({
+    lightingModel: Cesium.LightingModel.UNLIT,
+    vertexShaderText: `
+    void vertexMain(VertexInput vsInput, inout czm_modelVertexOutput vsOutput) {
+        vsOutput.pointSize = 4.0;
+    }
+    `,
+    fragmentShaderText: `
+    void fragmentMain(FragmentInput fsInput, inout czm_modelMaterial material) {
+        vec3 position = fsInput.attributes.positionMC;
+        vec3 octants = step(0.0, position);
+        material.diffuse = octants;
+    }
+    `
+});
+
+
+const VIEW_CLUSTER_COORDINATES = new Cesium.CustomShader({
+    lightingModel: Cesium.LightingModel.UNLIT,
+    vertexShaderText: `
+    void vertexMain(VertexInput vsInput, inout czm_modelVertexOutput vsOutput) {
+        vsOutput.pointSize = 4.0;
+    }
+    `,
+    fragmentShaderText: `
+    void fragmentMain(FragmentInput fsInput, inout czm_modelMaterial material) {
+        material.diffuse = fsInput.attributes.cluster_coordinates;
+    }
+    `
+});
+
+const ANIMATE_CUMULATIVE = new Cesium.CustomShader({
+    uniforms: {
+        u_iterations: {
+            type: Cesium.UniformType.FLOAT,
+            value: 1
+        },
+        u_time: {
+            type: Cesium.UniformType.FLOAT,
+            value: 0
+        }
+    },
+    lightingModel: Cesium.LightingModel.UNLIT,
+    vertexShaderText: `
+    void vertexMain(VertexInput vsInput, inout czm_modelVertexOutput vsOutput) {
+        float iter_normalized = vsInput.attributes.featureId_0 / (u_iterations - 1.0);
+        float t = mod(0.1 * u_time, 1.0);
+
+        vsOutput.pointSize = 4.0;
+
+        // hide points by multiplying by NaN
+        if (iter_normalized > t) {
+            vsOutput.positionMC = vec3(0.0) / 0.0;
+        }
+    }
+    `,
+
+    fragmentShaderText: `
+    void fragmentMain(FragmentInput fsInput, inout czm_modelMaterial material) {
+        float iter_normalized = fsInput.attributes.featureId_0 / (u_iterations - 1.0);
+        float t = mod(0.1 * u_time, 1.0);
+
+        float hotspot = smoothstep(0.2, 0.1, t - iter_normalized);
+        float brightness = smoothstep(0.5, 0.3, t - iter_normalized);
+        vec3 color = 0.8 * vec3(brightness) + 0.2;
+        color = mix(color, vec3(1.0, 0.5, 0.0), hotspot);
+        material.diffuse = color;
+    }
+    `
+});
+
+const ANIMATE_PULSE = new Cesium.CustomShader({
```

Comment:
this one will take a little more thinking. I need a function that looks like a bell curve (though goes from exactly 0 to 1) -- something like `smoothstep(1, 0, abs(x))` but moving in time and wraps smoothly at the edges. I'll have to play around with this a bit.